fvm/evm/stdlib/contract.go

```diff
@@ -1639,9 +1663,11 @@ func newInternalEVMTypeDeployFunction(
 
 			const isAuthorized = true
 			account := handler.AccountByAddress(fromAddress, isAuthorized)
-			address := account.Deploy(code, gasLimit, amount)
+			result := account.Deploy(code, gasLimit, amount)
 
-			return EVMAddressToAddressBytesArrayValue(inter, address)
+			// assign deployed contract address to return value since it's the only result of the deploy
+			result.ReturnedValue = result.DeployedContractAddress[:]
```

Comment:
I feel this logic should be part of the emulator.

Reply:
IMHO, this might complicate things, because EVM events rely on `types.Result` to generate their payload, e.g.

```go
cadence.NewUInt8(p.Result.TxType),
cadence.NewUInt64(p.Result.GasConsumed),
cadence.String(p.Result.DeployedContractAddress.String()),
cadence.String(hex.EncodeToString(p.Result.ReturnedValue)),
```

Special casing this assignment here, solely for the purpose of populating `EVM.Result` struct seems like a good way to go.